docs/internals/auth.md

```diff
@@ -1,7 +1,467 @@
+| <span style="background-color:teal;color:white">&nbsp;cache_dir&nbsp;</span> | string | * | null |
+
+Colored attributes are Fractal specific, all the others are [provided](https://github.com/fastapi-users/fastapi-users-db-sqlmodel/blob/main/fastapi_users_db_sqlmodel/__init__.py) by FastAPI Users.
+
+
+## Authentication
+
+### Login
+
+An _authentication backend_ is composed of two parts:
+
+- the <ins>transport</ins>, that manages how the token will be carried over the request,
+- the <ins>strategy</ins>, which manages how the token is generated and secured.
+
+Fractal Server provides two authentication backends, both using the [JWT](https://fastapi-users.github.io/fastapi-users/10.1/configuration/authentication/strategies/jwt/) strategy.<br>
+Each backend produces both [`/login`](https://fastapi-users.github.io/fastapi-users/12.1/usage/routes/#post-login) and [`/logout`](https://fastapi-users.github.io/fastapi-users/12.1/usage/routes/#post-logout) routes.
+
+> ⚠️ Since tokens are not stored in a database,`/logout` does not actually do anything.
+>
+> We have it because FastAPI Users always generates the two endpoints together.
+
+#### Bearer
+
+The [Bearer](https://fastapi-users.github.io/fastapi-users/10.1/configuration/authentication/transports/bearer/) transport backend provides login at `/auth/token/login`
+```console
+$ curl \
+    -X POST \
+    -H "Content-Type: application/x-www-form-urlencoded" \
+    -d "username=admin@fractal.xy&password=1234" \
+    http://127.0.0.1:8000/auth/token/login
+
+{
+    "access_token":"eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxIiwiYXVkIjpbImZyYWN0YWwiXSwiZXhwIjoxNjkzNTc1MzM1fQ.UmkhBKxgBM2mxXlrTlt5HXqtDDOe_mMYiMkKUS5jbXU",
+    "token_type":"bearer"
+}
+```
+
+#### Cookie
+
+The [Cookie](https://fastapi-users.github.io/fastapi-users/10.1/configuration/authentication/transports/cookie/) transport backend provides login at `/auth/login`
+
+```console
+$ curl \
+    -X POST \
+    -H "Content-Type: application/x-www-form-urlencoded" \
+    -d "username=admin@fractal.xy&password=1234" \
+    --cookie-jar - \
+    http://127.0.0.1:8000/auth/login
+
+
+#HttpOnly_127.0.0.1	FALSE	/	TRUE	0	fastapiusersauth	eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxIiwiYXVkIjpbImZyYWN0YWwiXSwiZXhwIjoxNjkzNjQ4MDI5fQ.UKRdbVjwys4grQrhpGyxcxcVbNSNJ29RQiFubpGYYUk
+```
+
+
+### Authenticated calls
+
+Once you have the token, you can use it to identify yourself by sending it along in the header of the request:
+
+```console
+$ curl \
+    -X GET \
+    -H "Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIyIiwiYXVkIjpbImZyYWN0YWwiXSwiZXhwIjoxNjkzOTI2MTM4fQ.MqWhW0xRgCV9ZgZr1HcdynrIJ7z46IBzO7pyfTUaTU8" \
+    http://127.0.0.1:8000/auth/whoami
+
+{
+    "id":1,
+    "email":"admin@fractal.xy",
+    "is_active":true,
+    "is_superuser":true,
+    "is_verified":false,
+    "slurm_user":null,
+    "cache_dir":null,
+    "username":"admin"
+}
+```
+
+
+## OAuth2
```

Comment:
As per our discussion, this section can be streamlined a bit.
Let's also mention the name of our OAuth2 flow explicitly, with a link (if useful).

Reply:
See https://github.com/fractal-analytics-platform/fractal-server/pull/822#issuecomment-1717683775.